src/cfml/system/util/CompileDSL.cfc

```diff
@@ -0,0 +1,402 @@
+********************************************************************************
+* @description
+*
+* I am a helper object for executing compile actions for "javac" command and
+* jarring actions for "jar" command. Create me and call my
+* methods to compile and jar any java project, then .run() will execute me.
+*
+*/
+
+component accessors=true {
+
+    /* root folder of the project you are working with (compiling, jar) */
+    property name='projectRoot'             type="string";
+    /* folder inside root project where source files reside (.java) */
+    property name='sourceDirectory'         type='string';
+    property name='classPathDirectory'      type='string';
+    property name='classOutputDirectory'    type='string';
+    property name='verbose'                 type='boolean';
+    property name='encode'                  type='string';
+	property name='sourcePaths'				type='array';
+	property name='recursive'				type='boolean';
+	property name='createJar'				type='boolean';
+	property name='jarNameString'			type='string';
+	property name='libsDir'					type='string';
+	property name='compileOptionsString'	type='string';
+	property name='jarOptionsString'		type='string';
+    property name='javaBinFolder'           type='string';
+
+    //DI
+	property name="packageService"	inject="PackageService";
+    property name='wirebox'         inject='wirebox';
+    property name='fileSystemUtil'  inject='FileSystem';
+	property name='shell'	        inject='shell';
+    property name='job'		        inject='interactiveJob';
+	property name="tempDir" 		inject="tempDir@constants";
+
+    /*
+    have a classpath
+    have a verbose flag
+    have a encoding flag option
+    */
+
+    public function init() {
+        setSourceDirectory( 'src\main\java\' );
+        setClassOutputDirectory( 'classes\java\main\' );
+        setVerbose( false );
+        setEncode( '' );
+		setSourcePaths( [''] );
+		setRecursive( false );
+		setCreateJar( false );
+		setLibsDir( 'libs' );
+		setCompileOptionsString( '' );
+		setJarOptionsString( '' );
+		setJarNameString( 'output.jar' );
+        return this;
+    }
+
+	/**
+	 * Sets the directory to run the command in
+  	 **/
+	function projectRoot( required projectRoot ) {
+		setProjectRoot( fileSystemutil.resolvePath( projectRoot ) );
+		return this;
+	}
+
+	function fromSource( required any sourcePaths ) {
+		if( isSimpleValue( arguments.sourcePaths ) ) {
+			arguments.sourcePaths = listToArray( arguments.sourcePaths, ",", true );
+		}
+		arguments.sourcePaths = arguments.sourcePaths.map( function( s ) {
+			return fileSystemutil.resolvePath( arguments.s, getProjectRoot() );
+		} );
+		variables.sourcePaths = arguments.sourcePaths;
+		return this;
+	}
+
+    function toClasses( required classOutputDirectory ){
+        setClassOutputDirectory( fileSystemutil.resolvePath( classOutputDirectory, getProjectRoot() ) )
+        return this;
+    }
+
+    function withVerbose() {
```

Comment:
Can we just call this method `verbose()`.  Methods like `withEncoding()` work but I think it's because Encoding is a noun and verbose is an adjective.  It's like saying an apple is 'withRed()`, it just doesn't flow as well.

Reply:
done